File: trunk/tests/CoreTests/OperatorOverloads/OperatorOverloads.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SharpKit.JavaScript;

namespace CoreTests.OperatorOverloads
{
    [JsType(JsMode.Prototype)]
    class Test
    {
        void foo()
        {
            var x = new BigNumber();
            var y = new BigNumber();
            x += y;
            var z = x + y;
            var a = MyBigNumber + x;
            MyBigNumber += a;
            
        }
        [JsProperty(NativeField=false)]
        public BigNumber MyBigNumber { get; set; }
    }
    [JsType(JsMode.Prototype)]
    class BigNumber
    {
        public static BigNumber operator +(BigNumber x, BigNumber y)
        {
            return new BigNumber();
        }
    }
}

File: tests/CoreTests/OperatorOverloads/OperatorOverloads.js

if (typeof(CoreTests) == "undefined")
    var CoreTests = {};
if (typeof(CoreTests.OperatorOverloads) == "undefined")
    CoreTests.OperatorOverloads = {};
CoreTests.OperatorOverloads.Test = function (){
    this._MyBigNumber = null;
};
CoreTests.OperatorOverloads.Test.prototype.foo = function (){
    var x = new CoreTests.OperatorOverloads.BigNumber();
    var y = new CoreTests.OperatorOverloads.BigNumber();
    x = CoreTests.OperatorOverloads.BigNumber.op_Addition(x, y);
    var z = CoreTests.OperatorOverloads.BigNumber.op_Addition(x, y);
    var a = CoreTests.OperatorOverloads.BigNumber.op_Addition(this.get_MyBigNumber(), x);
    this.set_MyBigNumber(CoreTests.OperatorOverloads.BigNumber.op_Addition(this.get_MyBigNumber(), a));
};
CoreTests.OperatorOverloads.Test.prototype.get_MyBigNumber = function (){
    return this._MyBigNumber;
};
CoreTests.OperatorOverloads.Test.prototype.set_MyBigNumber = function (value){
    this._MyBigNumber = value;
};
CoreTests.OperatorOverloads.BigNumber = function (){
};
CoreTests.OperatorOverloads.BigNumber.op_Addition = function (x, y){
    return new CoreTests.OperatorOverloads.BigNumber();
};

